File: Programmers/Level2/221025/JavaScript/RepeatBinaryTransformation.js
export function removeZero(sentence) {
  return sentence.replaceAll('0', '');
}

export function binaryTransformation(sentence) {
  const binary = sentence.toString(2);
  return binary;
}

export function solution(s) {
  const sentence = {
    initialSentence: s,
  };

  const answer = [];

  const count = {
    countZero: 0,
    countTransformation: 0,
  };

  while (sentence.initialSentence !== '1') {
    const newSentence = removeZero(sentence.initialSentence);
    count.countZero += sentence.initialSentence.length - newSentence.length;
    sentence.initialSentence = binaryTransformation(newSentence.length);
    count.countTransformation += 1;
  }

  answer[0] = count.countTransformation;
  answer[1] = count.countZero;

  return answer;
}

File: Programmers/Level2/221024/Java/RepeatBinaryTransformation.java
public class RepeatBinaryTransformation {
  public int[] solution(String s) {
    int[] answer = new int[2];

    int zeroCount = 0;
    int binaryCount = 0;

    while(s.contains("0") || s.length() != 1) {
      String newSentence = s.replaceAll("0", "");
      zeroCount += s.length() - newSentence.length();
      s = Integer.toBinaryString(newSentence.length());
      binaryCount += 1;
    }

    answer[0] = binaryCount;
    answer[1] = zeroCount;

    return answer;
  }
}
